File: utils/src/main/java/ru/virgil/utils/fluent_request/FinalStepByteType.java
package ru.virgil.utils.fluent_request;

import lombok.RequiredArgsConstructor;
import org.springframework.test.web.servlet.MvcResult;
import org.springframework.test.web.servlet.ResultMatcher;

@RequiredArgsConstructor
public class FinalStepByteType implements FinalStep<byte[]> {

    private final RequestModel requestModel;
    private final Requester requester;

    public byte[] expect(ResultMatcher... expectMatchers) throws Exception {
        requestModel.setResultMatchers(expectMatchers);
        MvcResult mvcResult = requester.makeRequest(requestModel);
        return mvcResult.getResponse().getContentAsByteArray();
    }
}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/testing/fluent_request/FinalStepByteType.kt
package ru.virgil.spring_tools.tools.testing.fluent_request

import org.springframework.test.web.servlet.MvcResult
import org.springframework.test.web.servlet.ResultMatcher

class FinalStepByteType(
    private val requestModel: RequestModel,
    private val requester: Requester,
) : FinalStep<ByteArray> {

    @Throws(Exception::class)
    override fun expect(vararg expectMatchers: ResultMatcher): ByteArray {
        requestModel.resultMatchers = (expectMatchers.toList())
        val mvcResult: MvcResult = requester.makeRequest(requestModel)
        return mvcResult.response.contentAsByteArray
    }
}
